server/tests/conftest.py

```diff
@@ -29,21 +30,43 @@
 # injection.
 
 
+# Reset the db once per test session. This means that every test will operate
+# with a shared db, which better simulates the real world.
+# Based on https://github.com/pytest-dev/pytest-xdist#making-session-scoped-fixtures-execute-only-once
+@pytest.fixture(scope="session", autouse=True)
+def reset_test_db(tmp_path_factory, worker_id):
+    # If we're not executing with multiple workers (from pytest-xdist), simply
+    # reset the db.
+    if not worker_id:
+        reset_db()
+
+    # Otherwise, use a file lock in a temp directory shared by all workers to
+    # ensure only one worker can reset the db.
+    root_tmp_dir = tmp_path_factory.getbasetemp().parent
+
+    temp_file = root_tmp_dir / "reset_db"
+    with FileLock(str(temp_file) + ".lock"):
+        if not temp_file.is_file():
+            reset_db()
+            temp_file.write_text("reset_db complete")
+
+
 @pytest.fixture
 def client() -> Generator[FlaskClient, None, None]:
     app.config["TESTING"] = True
     client = app.test_client()
 
-    reset_db()
+    # TODO run this once at the beginning of all tests?
+    # reset_db()
```

Comment:
We should probably remove these two lines ?

Reply:
Good call. I was actually able to simplify this fixture a lot now that I look at it: 260f44e